clang/lib/CodeGen/ItaniumCXXABI.cpp

```diff
@@ -3583,10 +3597,16 @@ void ItaniumRTTIBuilder::BuildVTablePointer(const Type *Ty) {
         llvm::ConstantExpr::getInBoundsGetElementPtr(CGM.Int8Ty, VTable, Eight);
   } else {
     llvm::Constant *Two = llvm::ConstantInt::get(PtrDiffTy, 2);
-    VTable = llvm::ConstantExpr::getInBoundsGetElementPtr(CGM.Int8PtrTy, VTable,
-                                                          Two);
+    VTable = llvm::ConstantExpr::getInBoundsGetElementPtr(
+        VTableName == ClassTypeInfo || VTableName == SIClassTypeInfo
+            ? CGM.Int8Ty->getPointerTo(/*addrspace =*/1)
+            : CGM.Int8PtrTy,
```

Comment:
This condition is being used in multiple places - factor it out and have a common routine for it?

Reply:
Or better make `GVType` available here and simply re-use it

Reply:
Done [341418c](https://github.com/intel/llvm/pull/7255/commits/341418c6205b89f553bfcc00eea5a5d4515e2cb8)